src/app/components/match-list/match-list.component.ts

```diff
@@ -1,22 +1,76 @@
 import { Component, OnInit, Input } from '@angular/core'
+import {animate, state, style, transition, trigger} from '@angular/animations';
+import { MatSort, MatTableDataSource } from '@angular/material'
+import { AuthenticationService } from '@services/authentication.service';
 import { Match } from '@models/match'
 import { Team } from '@models/team'
+import { Set } from '@models/set'
+
+interface MatchResult {
+  date: string,
+  homeTeam: string,
+  awayTeam: string,
+  score: string,
+}
 
 @Component({
   selector: 'app-match-list',
   templateUrl: './match-list.component.html',
-  styleUrls: ['./match-list.component.scss']
+  styleUrls: ['./match-list.component.scss'],
+  animations: [
+    trigger('detailExpand', [
+      state('collapsed', style({height: '0px', minHeight: '0'})),
+      state('expanded', style({height: '*'})),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4, 0.0, 0.2, 1)')),
+    ]),
+  ],
 })
 export class MatchListComponent implements OnInit {
 
   @Input() matches: Match[]
   @Input() teams: Team[]
 
-  constructor() { }
+  columnsToDisplay = ['date', 'homeTeam', 'awayTeam', 'score'];
+  dataSource: MatTableDataSource<MatchResult>;
+  expandedElement: MatchResult | null;
+
+  constructor(
+    private authService: AuthenticationService,
+  ) { }
 
   ngOnInit() { }
 
   getTeamNameById(id: number, teams: Team[]): string {
     return teams.find(team => team.id == id).name
   }
+
+  showReferee(id: number, teams:Team[]): string {
+    return id ? this.getTeamNameById(id, teams) : "No referee set"
+  }
+
+  getConcatinatedSets(sets: Set[]): string {
```

Comment:
Maybe this could be part of the result service? Maybe could be useful elsewhere.

Reply:
Yes, I think that sounds like a good idea. I have been quite unsure of how you've been using the different services vs the components logic. So my approach at the moment is that if I have to use it again I will move it